libcst/matchers/_visitors.py

```diff
@@ -296,13 +310,20 @@ def _gather_constructed_leave_funcs(
 
     for funcname in dir(obj):
         try:
-            func = cast(Callable[[cst.CSTNode], None], getattr(obj, funcname))
+            possible_func = getattr(obj, funcname)
+            if not ismethod(possible_func):
+                continue
+            func = cast(Callable[[cst.CSTNode], None], possible_func)
         except Exception:
```

Comment:
After adding the `not ismethod` check, do we still need this `except Exception:`?

Reply:
Yes, I think so. The exception isn't for the type, it is for if we accidentally do work on getattr due to properties.